drivers/include/periph/adc.h

```diff
@@ -30,6 +30,18 @@
  * waiting for the result of a conversion (e.g. through putting the calling
  * thread to sleep while waiting for the conversion results).
  *
+ * @section     sec_pm (Low-) power implications
+ *
+ * The ADC peripheral(s) **should** only be powered on while adc_read() is
+ * active. For implementing adc_read() this means, that the peripheral should be
+ * powered on (i.e. through peripheral clock gating) at the beginning of the
+ * function and it should be powered back off at the end of the function.
+ *
```

Comment:
No, the whole idea is implicit power saving (same concept as for `pm`): the adc is only powered on when actually used, so poweron/off are called implicitly.